keps/sig-network/1880-multiple-service-cidrs/README.md

```diff
@@ -0,0 +1,954 @@
+
+### New allocation model
+
+The new allocation mode requires:
+
+- 2 new API objects ServiceCIDRConfig and IPAddress in networking.k8s.io/v1alpha1, see <https://groups.google.com/g/kubernetes-sig-api-machinery/c/S0KuN_PJYXY/m/573BLOo4EAAJ>. Both will be protected with finalizers.
+- 1 new allocator implementing current `allocator.Interface`, that runs in each apiserver, and uses the new objects to allocate IPs for Services.
+- 1 new controller that participates on the ServiceCIDRConfig deletion to avoid that exist IPAddresses without
+  a ServiceCIDRConfig associated. It also handles the special case for the `default` ServiceCIDRConfig.
+- 1 new controller that participates in the IPAddress deletion, that guarantees that each Service IP
+  allocated has its corresponding IPAddress object, recreating it if necessary.
+- 1 new controller that handles the bootstrap process and the default ServiceCIDRConfig.
+
+[![](https://mermaid.ink/img/pako:eNp9UstqwzAQ_BWxx5IcCqUHUwrFacGXYJLeqh620tpV0SNIcsAk_vfKjpNiJ3QvYmdGs7NIBxBOEmQgNIawUlh7NNyyVAPCtuT3StDhhPWV6yZE8kXJQvTK1jeYwD4-52RRvqFRWlFPjk17Rbel00q07G7av7c7Omm7G-HyYrXJna1UfYm5RkOzfEW5f7iGHifQxL0oX6T0FMJ_rhuqmPv6ScfE4VynbszJONxzYE_H5fL4PDaXIVkCUGsnMFLgMLn4t-DcYj23EM5GVHZwgAUY8gaVTA88LMEhfpMhDr1UUoWNjr2yS9JmJ9PoV6mi85BVqAMtAJvotq0VkEXf0Fk0_pNR1f0CyVS2dw)](https://mermaid.live/edit#pako:eNp9UstqwzAQ_BWxx5IcCqUHUwrFacGXYJLeqh620tpV0SNIcsAk_vfKjpNiJ3QvYmdGs7NIBxBOEmQgNIawUlh7NNyyVAPCtuT3StDhhPWV6yZE8kXJQvTK1jeYwD4-52RRvqFRWlFPjk17Rbel00q07G7av7c7Omm7G-HyYrXJna1UfYm5RkOzfEW5f7iGHifQxL0oX6T0FMJ_rhuqmPv6ScfE4VynbszJONxzYE_H5fL4PDaXIVkCUGsnMFLgMLn4t-DcYj23EM5GVHZwgAUY8gaVTA88LMEhfpMhDr1UUoWNjr2yS9JmJ9PoV6mi85BVqAMtAJvotq0VkEXf0Fk0_pNR1f0CyVS2dw)
+
+#### The kube-apiserver bootstrap process and the service-cidr flags
+
+Currently, the Service CIDRs are configured independently in each kube-apiserver using flags. During
+the bootstrap process, the apiserver uses the first IP of each range to create the special
+"kubernetes.default" Service. It also starts a reconcile loop, that synchronize the state of the
+bitmap used by the internal allocators with the assigned IPs to the Services.
+
+With current implementation, each kube-apiserver can boot with different ranges configured.
+There is no conflict resolution, each apiserver keep writing and deleting others apiservers allocator bitmap
+and Services.
+
+In order to be completely backwards compatible, the bootstrap process will remain the same, the
+difference is that instead of creating a bitmap based on the flags, it will create a new
+ServiceCIDRConfig object from the flags (flags configuration removal is out of scope of this KEP)
+with a special label `networking.kubernetes.io/service-ip-cidr-default` set to `"true"`.
+
+The new bootstrap process will be:
+
+```
+at startup:
+ read_flags
+ if invalid flags
+  exit
+ run default-service-ip-range controller
+ run apiserver
+
+controller:
+ watch all ServiceCIDR objects labelled from-flags
```

Comment:
assuming this is the label above and you just changed what you're calling it?

Reply:
I'm asking for s/default/from-flags :)

Reply:
yes